ludwig/features/category_feature.py

```diff
@@ -352,6 +368,11 @@ def update_config_with_metadata(feature_config, feature_metadata, *args, **kwarg
         feature_config.num_classes = feature_metadata["vocab_size"]
         feature_config.top_k = min(feature_config.num_classes, feature_config.top_k)
 
+        # If labels are provided, then this is a classification task for LLMs
+        if hasattr(feature_config.preprocessing, "labels"):
```

Comment:
Let's actually call this `vocab` to be consistent with the cateogry distribution type. We can also make this generic such that ECD can also use it, and we just skip computing the str2idx stuff when that happens.

Reply:
@tgaddair are you suggesting renaming `labels` to `vocab`?

Reply:
We should probably consolidate on one. But it's okay to leave it as-is for now.